### PR TITLE
refs #6781 - add services for katello to services.d

### PR DESCRIPTION
**Pull request by stbenjam**

### CODE REVIEW

File: manifests/config.pp [hunk 1]
```diff
@@ -30,4 +30,6 @@
     mode   => '0755',
   }
 
+  $services_real = prefix($katello::services_list, $katello::services_directory)
+  ensure_resource('file', $services_real, {'ensure' => 'file', 'mode' => '0644'})
```
Comment (ekohl) [inline manifests/config.pp]: Any reason you're not simply using a file block?
```puppet
file { $services_real:
    ensure => 'file',
    mode   => '0644',
}
```

File: manifests/params.pp [hunk 2]
```diff
@@ -63,4 +63,15 @@
...
   $candlepin_event_queue = 'katello_event_queue'
+
+
+  if (versioncmp($::operatingsystemrelease, '7.0') < 0) {
+    $tomcat_service = 'tomcat6'
+  } else {
+    $tomcat_service = 'tomcat'
+  }
+
+  $services_directory = '/etc/katello/services.d'
```
Comment (ekohl) [inline manifests/params.pp]: Shouldn't this include a trailing slash since you're simplify prefixing in the code?

File: manifests/init.pp [hunk 3]
```diff
@@ -58,8 +63,12 @@
   $cdn_ssl_version = $katello::params::cdn_ssl_version,
 
   $package_names = $katello::params::package_names,
+
+  $services_directory = $katello::params::services_directory,
+  $services_list      = $katello::params::services_list,
   ) inherits katello::params {
 
+
```
Comment (ekohl) [inline manifests/init.pp]: Nitpick: why this additional line?

File: manifests/init.pp [hunk 4]
```diff
@@ -35,6 +35,11 @@
 #
 # $package_names::      Packages that this module ensures are present instead of the default: katello and ${scl_prefix}rubygem-katello
 #
+# $services_directory:: Directory where services configuration is stored
```
Comment (ekohl) [inline manifests/init.pp]: Can be validated with ```validate_absolute_path``` from stdlib.

File: manifests/init.pp [hunk 5]
```diff
@@ -35,6 +35,11 @@
...
+#
+# $services_list::      A list of services for the katello-service tool
```
Comment (ekohl) [inline manifests/init.pp]: Can be validated with ```validate_array``` from stdlib.

File: manifests/config.pp (hunk 1 above)
Comment (stbenjam) [inline manifests/config.pp]: It could be that puppet-katello and puppet-capsule end up with overlapping services.

Comment (ekohl) [inline manifests/config.pp]: In that case ensure_resource is dependant on the order (unless both use ensure_resource). Maybe a step back: why are these files needed even though they don't have any content (here)? In case you do want content, wouldn't a define be better? Similar to how we have ```foreman::plugin``` and then many ```foreman::plugin::*``` classes.

Comment (stbenjam) [inline manifests/config.pp]: They're both `ensure_resource`.

 `katello-service` tool will look services.d and restart all the services by the filename.  It's simple, but your comment makes me realize this approach doesn't work.

If you create say, `/etc/katello/services.d/pulp` on a capsule and later remove the service, the directory's not going to get cleared (and can't, it's managed by two modules).

I'll find a different approach

Comment (stbenjam) [inline manifests/config.pp]: Thanks @ekohl

**Comment by ehelms** [comment]
Do we still need this?

**Comment by stbenjam** [comment]
Nope
